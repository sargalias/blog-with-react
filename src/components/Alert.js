import React from 'react';

const Alert = ({ text, type }) => (
  <div
    className={`alert alert-${type} alert-dismissible fade show`}
    role="alert"
  >
    {text}
    <button type="button" className="close" data-dismiss="alert" aria-label="Close">
      <span aria-hidden="true">&times;</span>
    </button>
  </div>
);

export default Alert;
