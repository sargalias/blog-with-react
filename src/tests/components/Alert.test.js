import React from 'react';
import { shallow } from 'enzyme'
import Alert from '../../components/Alert';

test('alert should render correctly with no props', () => {
  const wrapper = shallow(<Alert />);
  expect(wrapper).toMatchSnapshot();
});

test('alert should render correctly with props', () => {
  const text = 'Test';
  const type = 'success';
  const wrapper = shallow(<Alert text={text} type={type} />);
  expect(wrapper).toMatchSnapshot();
});

test('alert should use its text prop correctly', () => {
  const text = "Test";
  const matcher = /^Test/;
  const wrapper = shallow(<Alert text={text} />);
  expect(wrapper.text()).toMatch(matcher);
});

test('alert should use its type prop properly', () => {
  const type = 'test';
  const wrapper = shallow(<Alert type={type} />);
  expect(wrapper.hasClass(`alert-${type}`)).toBe(true);
});
