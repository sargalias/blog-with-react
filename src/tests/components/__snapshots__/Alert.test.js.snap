// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alert should render correctly with no props 1`] = `
<div
  className="alert alert-undefined alert-dismissible fade show"
  role="alert"
>
  <button
    aria-label="Close"
    className="close"
    data-dismiss="alert"
    type="button"
  >
    <span
      aria-hidden="true"
    >
      ×
    </span>
  </button>
</div>
`;

exports[`alert should render correctly with props 1`] = `
<div
  className="alert alert-success alert-dismissible fade show"
  role="alert"
>
  Test
  <button
    aria-label="Close"
    className="close"
    data-dismiss="alert"
    type="button"
  >
    <span
      aria-hidden="true"
    >
      ×
    </span>
  </button>
</div>
`;
